docs/section-7/drivetrain-characterization.md

```diff
@@ -0,0 +1,123 @@
+
+### Computing the Constants
+
+Launch the Data Analyzer using the Launch Data Analyzer button on the original window. There should be a new pop-up where you can select the data file that you created in the previous section.
+
+Before you hit the Analyze Data Button, make sure that you change your wheel diameter in the top-right corner to use the correct units and is the right value. Also, make sure that you change the subset value so that you can gather data for forward left, forward right, etc. Then, hit analyze data.
+
+Make sure to record the kS (\(k_{volts})\), kV, and kA values for each side and each direction. These values can be found in the top-left corner of the Feedforward Analysis section. As well, you will need to record the kP and (depending on application) kD for the left and right sides. These values can be found in the bottom-right corner of the Feedback Analysis section
+
+## Implementing Drivetrain Characterization
+
+From here, implementation is not very difficult, however, the code is kind of dense.
+
+You will need to initialize six objects: 1 ```SimpleMotorFeedforward``` objects for each side and direction (4 total), and 1 ```PIDController``` for each side. The constructor for each ```SimpleMotorFeedforward``` will require the kS, kV, and kA we found in the previous section and each ```PID Controller``` will require the kP (and possibly kD) from the previous section.
+
+Our characterized driving code will require two parameters, left and right, the desired speed of the wheels in m/s for the left and right sides. The code will then supply the motors with the correct voltage. Below are the instructions for creating our characterized driving code:
+
+1. For the left and right sides, determine the desired acceleration of the robot by finding the change in velocity between our desired velocity and our current velocity and dividing by the change in time between the two measurements.
+
+2. (Optional) Clip the acceleration at some maximum value. Note that the maximum acceleration the robot can achieve without the wheel slipping is \(\mu_s g\) where \(\mu_s\) is the coefficient of static friction and \(g\) is the gravitational acceleration.
+
+3. Apply the drivetrain characterization formula for each side and direction using the ```SimpleMotorFeedforward.calculate()``` method.
+
+4. Determine the adjustment value for each side using the ```PIDController.calculate()``` method.
+
+5. If the robot will be moving forward on the left side, the desired voltage is the sum of the left PID output and the forward left FF output; otherwise, the voltage is the sum of the left PID output and the backward left FF output; otherwise, the voltage. Repeat for the right side.
+
+6. Divide by the maximum voltage and supply the final results to your drivetrain's ```tankDrive()``` method.
+
+You should now have everything you need to implement your own code for a characterized drivetrain.
```

Comment:
Is there stuff here that isn't covered in the [official docs](https://docs.wpilib.org/en/latest/docs/software/wpilib-tools/robot-characterization/index.html)? I recommend referring the reader to those docs and just adding any tips/tricks that aren't covered there.